mcs/class/Microsoft.Build.Engine/Test/Microsoft.Build.BuildEngine/ProjectTest.cs

```diff
@@ -1301,6 +1301,34 @@ public void TestBuildProjectFileXml8 ()
 		}
 
 		[Test]
+		public void TestBatchedMetadataEmptyBatch () {
```

Comment:
I tried this test without this PR and it passed there as well, so it's not really testing for the right problem.